### PR TITLE
Feature/v12 compatibility 1

### PR DESCRIPTION
**Pull request by maikschneider**
I activated the code in `GenerateRequestTokenListener.php` and did little adjustments `BackendAuthenticationService.php`. Now the extension is working in my v12.4.6.

Needs further tests and probably a security check regarding the request token stuff.

### CODE REVIEW

File: Classes/Authentication/BackendAuthenticationService.php [hunk 1]
```diff
@@ -35,7 +35,7 @@
 use Waldhacker\Oauth2Client\Service\Oauth2Service;
 use Waldhacker\Oauth2Client\Session\SessionManager;
 
-class BackendAuthenticationService
+class BackendAuthenticationService extends AbstractAuthenticationService
```
Comment (Loeffeldude) [inline Classes/Authentication/BackendAuthenticationService.php:38]: This also needs to be done for the FrontendAuthenticationService

File: Classes/Events/Listener/GenerateRequestTokenListener.php [hunk 2]
```diff
@@ -35,10 +35,9 @@ public function __invoke(BeforeRequestTokenProcessedEvent $event): void
         // wenn oauth cookie => get time from token from cookie
         // wenn request within 5 sec => generate and set token
 
-        // $requestToken = RequestToken::create('core/user-auth/be');
-        // $event->setRequestToken($requestToken);
+        $requestToken = RequestToken::create('core/user-auth/be');
```
Comment (Loeffeldude) [inline Classes/Events/Listener/GenerateRequestTokenListener.php]: This needs to also be adjusted for frontend.

```suggestion
        $user = $event->getUser();
        $requestToken = RequestToken::create('core/user-auth/' . strtolower($user->loginType));
```

**Review by Loeffeldude — COMMENTED** [review]
This is still broken for frontend. I got it working with these 2 minor fixes.

**Comment by maikschneider** [comment]
Thank you @Loeffeldude!

**Comment by waldhacker1** [comment]
Thanks for your contribution!
The Repository has been moved to [GitLab](https://gitlab.com/co-stack.com/co-stack.com/typo3-extensions/typo3-oauth2-client).
Please consider to push your code there.

**Pull request closed without merging**